server/server.js

```diff
@@ -7,158 +7,97 @@ const db = require('./db/db-connection.js');
+      });
+  } catch (e) {
+    return res.status(400).json({ e });
+  }
+});
 
-// create the post request for users in the endpoint '/api/users'
+//get request for users in the endpoint '/api/user'
+app.get('/api/user', cors(), async (req, res) => {
+  try {
+    const { rows: users } = await db.query("SELECT * FROM users");
+    res.send(users);
+  } catch (e) {
+    return res.status(400).json({ e });
+  }
+});
+
+//post request for user in the endpoint '/api/user'
 app.post('/api/user', async (req, res) => {
-    try {
-        const userAccount = req.body.user;
-        const userEmail = await db.query("SELECT * from users where email = $1", [userAccount.email,])
-        if (userEmail.rows.length === 0){
-            const newUser = {
-                username : req.body.user.name,
-                email: req.body.user.email,
-            };
-            const result = await db.query("INSERT INTO users(username, email) VALUES ($1,$2) RETURNING *",
-            [newUser.username, newUser.email]
-          );
-        }
-        console.log("line34", result.rows[0]);
-        res.json(result.rows[0]);
-    } catch (e) {
-        return res.status(400).json({ e });
+  try {
+    const userAccount = req.body.user;
+    const queryResult = await db.query("SELECT * from users where email = $1", [userAccount.email,])
+    if (queryResult.rows.length === 0) {
+      const newUser = {
+        username: req.body.user.name,
+        email: req.body.user.email,
+      };
+      const result = await db.query("INSERT INTO users(username, email) VALUES ($1,$2) RETURNING *",
+        [newUser.username, newUser.email]
+      );
+    } else {
+      let result = queryResult
     }
+    res.json(result.rows[0]);
+  } catch (e) {
+    return res.status(400).json({ e });
+  }
 });
 
-// app.get("/api/user/favorites/:email", cors() , async (req, res) => {
-//   try {
-//     const { email }  = req.params;
-//     const { rows: favorites } = await db.query("SELECT * f.userid FROM favorites f JOIN users u ON u.id = f.id WHERE email = $1",
-//     [email]);
-//     res.send(favorites);
-//   } catch (e) {
-//     return res.status(400).json({ e });
-//   }
-// });
-app.get("/api/user/favorites/:email", cors() , async (req, res) => {
+//get request for user favorites in the endpoint '/api/user/favorites/:email'
+app.get("/api/user/favorites/:email", cors(), async (req, res) => {
   try {
-    const { email }  = req.params;
+    const { email } = req.params;
     const { rows: favorites } = await db.query("SELECT pokecode FROM favorites JOIN users ON favorites.userid = users.id WHERE email = $1",
-    [email]);
+      [email]);
     res.send(favorites);
   } catch (e) {
     return res.status(400).json({ e });
   }
 });
 
-app.post('/api/addFavorite/:pokecode', async (req, res) => {
+//post request for user favorite in the endpoint '/api/addFavorite/:pokecode/:email'
+app.post('/api/addFavorite/:pokecode/:email', async (req, res) => {
```

Comment:
This works. In most apps, what you will usually see is devs using the user's database id instead of the email as a param. It protects the user's email address from being exposed to anyone snooping on the requests. The user id is also presumed to be unique, whereas you may potentially have multiple users with the same email address.